docs/docs/maintaining-a-plugin.md

```diff
@@ -1,10 +1,50 @@
 ---
 title: Maintaining a Plugin
-issue: https://github.com/gatsbyjs/gatsby/issues/21067
 ---
 
-This is a stub article meant to be filled with tips on how to maintain a Gatsby plugin once you've published it as an npm package.
+The Gatsby community thrives on the power of plugins, you often find that there is a plugin for almost everything ranging from source plugins to transformer plugins to plugins that are plugins for other plugin . There is a guide to for the [naming convention for plugins](https://www.gatsbyjs.org/docs/naming-a-plugin/) check it out to learn more .
 
-Topics to be covered:
+> Wondering how to [Creating a plugin](https://www.gatsbyjs.org/docs/creating-plugins/) ? Look no further . Start contributing .
 
-- yarn workspaces can solve yarn link inconsistencies
+## What is a Plugin ?
+
+Gatsby plugins are Node.js packages that implement Gatsby APIs. [Learn more about Plugins](https://www.gatsbyjs.org/docs/plugins/) in this part of the Gatsby documentation.
+
+The bulk of the work is during the development phase. However, there’s still a need to take a look at the dependencies and security features from time to time. Keeping Plugins up to date becomes really tasking and can sometimes become really complicated since updating dependencies could potentially break your apps.
+
+However, the effects of not updating and maintaining can bring a huge security issue to your apps and we don't want that. Let's look at some tips to apply when maintaining plugins:
+
+## Handling patches and improvements
+
+Most plugins generally follow [Semantic Versioning](https://semver.org/) to determine how many releases to do and the type of releases.
+
+The first public release of a plugin is 0.1.0 or 1.0.0. From there, bugs are fixed in patch releases (e.g. 0.1.1) features are added or changed in minor releases (e.g. 0.2.0).
+
+Version 1.0.0 should be released when the plugin's API is considered stable. Version 2.0.0 (and subsequent major releases) would introduce breaking API changes.
+
+> Consider a version format of X.Y.Z (Major.Minor.Patch). Bug fixes not affecting the API increment the patch version, backwards compatible API additions/changes increment the minor version, and backwards incompatible API changes increment the major version.
+
+## Update ReadMe and Use cases
+
+Every major version update should also be reflexed in the READme of the Plugin and also in the use case examples.
+
+It would be great for users to be able reference several versions of the plugin with the updated examples to see if they want to keep the current version or upgrade and also to understand what the new version offers. Although this is good:
```

Comment:
```suggestion
It would be great for users to be able to reference several versions of the plugin with the updated examples to see if they want to keep the current version or upgrade and also to understand what the new version offers. Although this is good:
```